services/brig/src/Brig/API/Client.hs

```diff
@@ -328,6 +329,52 @@ claimMultiPrekeyBundles protectee quc = do
       tUntagged . qualifyAs luc
         <$> claimLocalMultiPrekeyBundles protectee (tUnqualified luc)
 
+-- Similar to claimMultiPrekeyBundles except for the following changes
+-- 1) A new return type that contains both the client map and a list of
+--    users that messages couldn't be sent to.
+-- 2) A semantic change on federation errors when gathering remote clients.
+--    Remote federation errors at this step no-longer cause the entire call
+--    to fail, allowing partial results to be returned.
+claimMultiPrekeyBundlesV3 ::
+  forall r.
+  Member (Concurrency 'Unsafe) r =>
+  LegalholdProtectee ->
+  QualifiedUserClients ->
+  ExceptT ClientError (AppT r) QualifiedUserClientPrekeyMapV3
+claimMultiPrekeyBundlesV3 protectee quc = do
```

Comment:
If the old handler cannot be repurposed to handle both the old and the new endpoint, can you factor out the common bits? My hunch is they should be quite similar, except for the remote part and the construction of the response.

Reply:
Done and done